src/data/repositories/data-entry/AMCProductDataDefaultRepository.ts

```diff
@@ -73,25 +74,21 @@ export class AMCProductDataDefaultRepository implements AMCProductDataRepository
 
     // TODO: decouple TrackerPostResponse from DHIS2
     importCalculations(
+        importStrategy: ImportStrategy,
         productDataTrackedEntities: ProductDataTrackedEntity[],
         rawSubstanceConsumptionCalculatedStageMetadata: ProgramStage,
         rawSubstanceConsumptionCalculatedData: RawSubstanceConsumptionCalculated[],
-        orgUnitId: Id,
-        orgUnitName: string
+        orgUnitId: Id
     ): FutureData<TrackerPostResponse> {
         const d2TrackerEvents = this.mapRawSubstanceConsumptionCalculatedToD2TrackerEvent(
             productDataTrackedEntities,
             rawSubstanceConsumptionCalculatedStageMetadata,
             rawSubstanceConsumptionCalculatedData,
             orgUnitId,
-            orgUnitName
+            importStrategy
         );
         if (!_.isEmpty(d2TrackerEvents)) {
-            return importApiTracker(
-                this.api,
-                { events: d2TrackerEvents },
-                TRACKER_IMPORT_STRATEGY_CREATE_AND_UPDATE
-            ).flatMap(response => {
+            return importApiTracker(this.api, { events: d2TrackerEvents }, importStrategy).flatMap(response => {
                 return Future.success(response);
```

Comment:
I guess there was some code before, but as it stand, flatMap+success is a no-op, it can be removed.